### PR TITLE
Added new SegmentSplitter executor

### PR DESCRIPTION
**Pull request by Ercadio**

### CODE REVIEW

File: src/data/BBFields.py [hunk 1]
```diff
@@ -235,5 +235,7 @@ def set_fields_from_dict(self, d):
                         != len(getattr(self, param)):
                     raise InvalidData(f"Length of all arrays must be equal")
             else:
-                raise InvalidData(
-                    f"Invalid Bounding-Box, param: {param} missing")
+                setattr(self, param, [])
+                pprint(d)
```
Comment (Ercadio) [inline src/data/BBFields.py:239]: Should get rid of this line and the exception comment

File: src/data/MetaDataItem.py [hunk 2]
```diff
@@ -24,6 +24,7 @@ def __init__(self, **kwargs):
         bb_fields_json = kwargs.get("bb_fields", {})
         if accident_locations:
             bb_fields_json["accident_locations"] = accident_locations
+        print(f'Parsing item {self.id}')
```
Comment (Ercadio) [inline src/data/MetaDataItem.py:27]: Should get rid of this

**Comment by Ercadio** [comment]
Marked WIP because I haven't properly tested that it works, and I need to remove the equivalent code in CsvExporter

File: src/executor/CsvExporter.py [hunk 3]
```diff
@@ -97,31 +66,26 @@ def run(self, item: VideoItem):
         data = np.array(np.split(data, unique_indices[1:], axis=0))
         
         # Mask for downsampling fps
-        n_input_frames  = collision_frame - begin
+        n_input_frames  = end - begin
```
Comment (Ercadio) [inline src/executor/CsvExporter.py:69]: Pretty sure we are missing a +1 here

**Review by Ercadio — COMMENTED** [review]
Also should update the requirements.txt to include ffmpeg. Not sure how it will impact release...

File: src/executor/CsvExporter.py [hunk 4]
```diff
@@ -20,74 +19,44 @@
...
     def run(self, item: VideoItem):
         print("Start exporting file")
         metadata = iExecutor.get_metadata(item)
-        bbs = metadata.bb_fields.to_json()
+        fps = item.fps
+        bbs = metadata.bb_fields.get_bbs_as_arrs()
         if bbs is None or len(bbs) == 0:
-            raise StopSignal("No bounding box fields for this item")
+            raise SkipSignal("No bounding box fields for this item")
+        accident_locations = metadata.bb_fields.accident_locations
```
Comment (nicholas-geofroy) [inline src/executor/CsvExporter.py:33]: should be collision_locations now

File: src/executor/SegmentSplitter.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        super().__init__(*parents)
+        self.clip_len_s = SegmentSplitter.parse_time(clip_length)
+        self.len_thresh_s = SegmentSplitter.parse_time(length_threshold)
+    
+    def split_segment(self, item):
+        metadata = iExecutor.get_metadata(item)
+        # First we find the length of BBs
+        bbs = metadata.bb_fields.get_bbs_as_arrs()
+        fps = item.fps
+        accident_locations = metadata.bb_fields.accident_locations
```
Comment (nicholas-geofroy) [inline src/executor/SegmentSplitter.py:17]: should be collision_locations

File: src/data/VideoFile.py [hunk 6]
```diff
@@ -92,11 +93,23 @@ def next(self) -> np.ndarray:
...
                 self.current_frame = frame
+                self.current_time = self.capture.get(cv2.CAP_PROP_POS_MSEC)
 
         return self.current_frame
 
     def current(self):
         return self.current_frame
+    
+    def current_time(self):
+        return self.current_time
```
Comment (nicholas-geofroy) [inline src/data/VideoFile.py:104]: does this work? we have the function current_time and the variable which have the same name?

**Review by nicholas-geofroy — APPROVED** [review]

File: src/data/VideoFile.py (hunk 6 above)
Comment (Ercadio) [reply src/data/VideoFile.py:104]: I agree it's confusing. I will just keep the field

**Pull request merged**